File: app/src/main/java/ai/elimu/analytics/entity/StoryBookLearningEvent.java
package ai.elimu.analytics.entity;

import androidx.annotation.NonNull;
import androidx.room.Entity;

import ai.elimu.model.v2.enums.analytics.LearningEventType;

@Entity
public class StoryBookLearningEvent extends LearningEvent {

    @NonNull
    private Long storyBookId;

    private String storyBookTitle;

    @NonNull
    private LearningEventType learningEventType;

    public Long getStoryBookId() {
        return storyBookId;
    }

    public void setStoryBookId(Long storyBookId) {
        this.storyBookId = storyBookId;
    }

    public String getStoryBookTitle() {
        return storyBookTitle;
    }

    public void setStoryBookTitle(String storyBookTitle) {
        this.storyBookTitle = storyBookTitle;
    }

    public LearningEventType getLearningEventType() {
        return learningEventType;
    }

    public void setLearningEventType(LearningEventType learningEventType) {
        this.learningEventType = learningEventType;
    }
}

File: app/src/main/java/ai/elimu/analytics/entity/StoryBookLearningEvent.kt
package ai.elimu.analytics.entity

import ai.elimu.model.v2.enums.analytics.LearningEventType
import androidx.room.Entity

/**
 * Based on https://github.com/elimu-ai/webapp/blob/main/src/main/java/ai/elimu/entity/analytics/StoryBookLearningEvent.java
 */
@Entity
class StoryBookLearningEvent : LearningEvent() {
    lateinit var storyBookTitle: String

    var storyBookId: Long = 0L

    @Deprecated("Will be replaced by `additionalData`")
    var learningEventType: LearningEventType? = null
}
